Strided masked norm.

pearu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #68738
* #68694
* __->__ #68584

Differential Revision: [D32581285](https://our.internmc.facebook.com/intern/diff/D32581285)

pearu committed:
Strided masked norm.

[ghstack-poisoned]

pearu committed:
Update on "Strided masked norm."




[ghstack-poisoned]

pearu committed:
Update on "Strided masked norm."




[ghstack-poisoned]

pearu committed:
Update on "Strided masked norm."




[ghstack-poisoned]

cpuhrsch commented:
@cpuhrsch has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32581285).

pearu committed:
Update on "Strided masked norm."


Differential Revision: [D32581285](https://our.internmc.facebook.com/intern/diff/D32581285)

[ghstack-poisoned]

cpuhrsch commented on test/test_masked.py:
```diff
@@ -12,6 +12,47 @@
     (instantiate_device_type_tests, ops, onlyNativeDeviceTypes)
 
 
+def apply_masked_reduction_along_dim(op, input, *args, **kwargs):
```
Can you add some comments to the code? I find it difficult to follow. In particular the pieces around "dim_position" and ranges.

pearu replied:
Extended the docstring with the description of the algorithm and commented on the code.

cpuhrsch reviewed (changes requested):
See comment

pearu committed:
Update on "Strided masked norm."


Differential Revision: [D32581285](https://our.internmc.facebook.com/intern/diff/D32581285)

[ghstack-poisoned]

cpuhrsch commented:
@cpuhrsch has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32581285).
